src/db.py

```diff
@@ -86,9 +94,28 @@ def overwrite_student_roster(cid, student_ids):
 	return mongo.db.courses.update({"_id": cid}, {"$set": {"student_ids": student_ids}})
 
 
-def get_assignments_for_course(cid, visible_only=True):
+def get_assignments_for_course(cid, visible_only=False):
 	if visible_only:
-		return list(mongo.db.assignments.find({"course_id": cid, "visibility": True}))
+		now = util.now_timestamp()
+		return list(mongo.db.assignments.find({
+			"course_id": cid,
+			"$or": [
+				# if visible from start date, and start date has past
+				{
+					"visibility": Visibility.VISIBLE_FROM_START,
+					"start": {
+						"$lte": now
+					}
+				},
+				{
+					"visibility": Visibility.VISIBLE
+				},
+				# TODO take this out after database shift
```

Comment:
We can remove this comment now.

Reply:
We still need this comment, we will remove this when we fix the prod DB.